### PR TITLE
[RHOAIENG-2948] Improve create job

### PR DESCRIPTION
**Pull request by andrewballantyne**
https://issues.redhat.com/browse/RHOAIENG-2948

## Description

Improved the create job flow to include a couple fields that we were missing (catch up & max concurrency). Reworked the start/end to be more aligned with the mocks and converted the radio buttons into a dropdown.

cc @yannnz / @xianli123 / @kywalker-rh 

https://github.com/opendatahub-io/odh-dashboard/assets/8126518/64471d85-a554-4dbe-bb4e-2ff47d7f7f0d

## How Has This Been Tested?

Create / Duplicate Runs & Jobs

## Test Impact

Added Tests for the scheduled section

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/__tests__/cypress/cypress/support/commands/application.ts [hunk 1]
```diff
@@ -133,3 +147,13 @@ Cypress.Commands.add('fill', { prevSubject: 'optional' }, (subject, text, option
...
+
+Cypress.Commands.add('pfSwitch', (dataId) => {
+  Cypress.log({ displayName: 'pfSwitch', message: dataId });
+  return cy.findByTestId(dataId).parent();
+});
+
+Cypress.Commands.add('pfSwitchValue', (dataId) => {
+  Cypress.log({ displayName: 'pfSwitchValue', message: dataId });
+  return cy.pfSwitch(dataId).find('[type=checkbox]');
+});
```
Comment (andrewballantyne) [inline frontend/src/__tests__/cypress/cypress/support/commands/application.ts:159]: @christianvogt thoughts?

Comment (christianvogt) [reply frontend/src/__tests__/cypress/cypress/support/commands/application.ts:159]: My first thought is by not do `cy.pfSwitch('id').switchValue()...`. I tried a quick solution but I need more experience with custom commands and specifically using typescript to create better types for such commands. I would want `switchValue` to only be available when chaining from `pfSwitch`.

This works for now and we can revisit later.

File: frontend/src/concepts/pipelines/content/createRun/contentSections/RunTypeSectionScheduled.tsx [hunk 2]
```diff
@@ -31,98 +15,14 @@ type RunTypeSectionScheduledProps = {
...
-                      })
-                    }
-                  />
-                </SplitItem>
-              </Split>
-            </>
-          )
-        }
+      <MaxConcurrencyField
+        onChange={(maxConcurrency) => onChange({ ...data, maxConcurrency })}
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/createRun/contentSections/RunTypeSectionScheduled.tsx:22]: I am able to set the value to 0 which throws an error on submit. not by arrow but by typing

Comment (Gkrumbach07) [reply frontend/src/concepts/pipelines/content/createRun/contentSections/RunTypeSectionScheduled.tsx:22]: same thing with setting large numbers over 10

**Comment by kywalker-rh** [comment]
lgtm based on the video and Yan's mocks

File: frontend/src/concepts/pipelines/content/createRun/contentSections/RunTypeSectionScheduled.tsx (hunk 2 above)
Comment (andrewballantyne) [reply frontend/src/concepts/pipelines/content/createRun/contentSections/RunTypeSectionScheduled.tsx:22]: Thanks for catching that, fixed and added a test for it.

**Review by Gkrumbach07 — APPROVED** [review]
/lgtm

**Pull request merged**